### ISSUE TITLE
Strange error on freebsd in Downloads test

### ISSUE THREAD

**Issue by Keno** [JuliaLang/julia#39582]
```
      From worker 3:	  Test threw exception
      From worker 3:	  Expression: download("https://httpbin.julialang.org/ip", file) == file
      From worker 3:	  could not load symbol "uv_poll_start":
      From worker 3:	  Shared object "nss_dns.so.1" not found, required by "julia"
      From worker 3:	  Stacktrace:
```

https://build.julialang.org/#/builders/4/builds/502/steps/5/logs/stdio

**Comment by vtjnash** [comment]
This might be basically FreeBSD being brain-dead. The error here looks like what would happen if they didn't make dlsym to be thread-safe. Note that the the standard doesn't actually mandate that libdl be thread-safe, so unlike everybody else, who does the obvious thing and makes it thread-safe, they don't document that it is threads-safe. It thus possible that no application may ever safely use dlopen/dlsym/dlerror without risk of process corruption. You can't add your own locks either, since their libc assumes it is thread-safe, and happily uses it from many threads (which is what happened here). This  might be https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=95339 or https://austin-group-l.opengroup.narkive.com/JI4xEbeU/thread-safe-dlerror, in which FreeBSD talked many years about about how the current situation might be bad.

**Comment by vtjnash** [comment]
Yeah, looks like they close that issue in 2018, since it was "believed fixed", but here's the obviously broken code still present https://github.com/freebsd/freebsd-src/blob/12db51d20823a5e3b9e5f8a2ea73156fe1cbfc28/libexec/rtld-elf/rtld.c#L198

@ararslan as a freebsd user, would you be willing to make an issue on the above issue tracker link, that `dlerror` is un-thread-safe?

**Comment by ararslan** [comment]
> would you be willing to make an issue on the above issue tracker link, that `dlerror` is un-thread-safe?

[Done](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=95339#c5)

**Comment by vtjnash** [comment]
Appears to be now fixed upstream also
